pkg/bff/gds.go

```diff
@@ -404,5 +425,12 @@ func (s *Server) SubmitRegistration(c *gin.Context) {
 		return
 	}
 
+	// Commit the user metadata updates to auth0
+	if err = s.SaveAuth0AppMetadata(*user.ID, *appdata); err != nil {
+		log.Error().Err(err).Str("user_id", *user.ID).Msg("could not save user app metadata")
+		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not save user app metadata"))
```

Comment:
If this fails, maybe it's okay to just fall back to the current state, which is that the user will have to log out and log back in again (rather than erroring with a 500 here). No matter what, the GDS registration has already been submitted at this point and we don't want them to misunderstand and believe they need to resubmit. We should keep the error log message though so we can introspect!

Reply:
That's a good point, I'll keep the error to make sure we catch that!